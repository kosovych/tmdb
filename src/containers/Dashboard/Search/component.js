import React from 'react'
import {
  Row, Col, Input
} from 'antd'
import PropTypes from 'prop-types'
import { Form, Field } from 'formik'
import InputField from 'Components/InputField'
import ClearButton from './ClearButton'


const Search = ({
  handleSubmit, values, onClear, handleBlur
}) => (
  <Row
    justify="center"
    gutter={{
      xs: 8, sm: 16, md: 24, lg: 22
    }}
  >
    <Col
      className="gutter-row"
      xs={{ span: 20 }}
      sm={{ span: 20 }}
      md={{ span: 14 }}
      lg={{ span: 12 }}
      xl={{ span: 10 }}
    >
      <Form onSubmit={handleSubmit}>
        <Field
          component={InputField}
          inputComponent={Input.Search}
          name="search"
          placeholder="Enter movie name"
          size="large"
          enterButton="Search"
          className="top-margin"
          onSearch={handleSubmit}
          onBlur={values.search && handleBlur}
          suffix={(
            <ClearButton
              onClick={onClear}
              isVisible={values.search}
            />
          )}
        />
      </Form>
    </Col>
  </Row>
)

Search.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  values: PropTypes.shape({
    search: PropTypes.string.isRequired
  }).isRequired,
  onClear: PropTypes.func.isRequired,
  handleBlur: PropTypes.func.isRequired
}

export default Search
