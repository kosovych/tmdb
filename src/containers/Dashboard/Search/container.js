/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react'
import { withFormik } from 'formik'
import * as Yup from 'yup'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { getMovies as getMoviesAction, setSearch as setSearchAction } from 'Store/trendingMovies/actions'
import SearchComponent from './component'

const SearchSchema = Yup.object().shape({
  search: Yup.string().min(2, 'Search should contain at least 2 characters').required()
})

class Search extends Component {
  onClear = () => {
    const {
      handleReset, setFieldTouched, submitCount, getMovies, setSearch
    } = this.props
    setFieldTouched('search', false)
    if (submitCount > 0) {
      setSearch('')
      getMovies()
    }
    handleReset()
  }

  render() {
    return (
      <SearchComponent
        {...this.props}
        onClear={this.onClear}
      />
    )
  }
}

Search.propTypes = {
  handleReset: PropTypes.func.isRequired,
  setFieldTouched: PropTypes.func.isRequired,
  submitCount: PropTypes.number.isRequired,
  getMovies: PropTypes.func.isRequired,
  setSearch: PropTypes.func.isRequired
}

const mapDispatchToProps = {
  getMovies: getMoviesAction,
  setSearch: setSearchAction
}

export default connect(null, mapDispatchToProps)(withFormik({
  mapPropsToValues: () => ({ search: '' }),
  handleSubmit: (values, { props }) => {
    const { getMovies, setSearch } = props
    setSearch(values.search)
    getMovies()
  },
  validationSchema: SearchSchema
})(Search))
