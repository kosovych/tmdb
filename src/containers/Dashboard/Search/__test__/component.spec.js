import React from 'react'
import { shallow } from 'enzyme'
import Search from '../component'

describe('Search component', () => {
  const props = {
    handleSubmit: jest.fn(),
    values: {
      search: 'Search...'
    },
    onClear: jest.fn(),
    handleBlur: jest.fn()
  }
  it('should match snapshot', () => {
    const component = shallow(<Search {...props} />)
    expect(component).toMatchSnapshot()
  })
})
