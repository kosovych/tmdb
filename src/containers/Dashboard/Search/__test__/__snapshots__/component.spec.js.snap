// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should match snapshot 1`] = `
<Row
  gutter={
    Object {
      "lg": 22,
      "md": 24,
      "sm": 16,
      "xs": 8,
    }
  }
  justify="center"
>
  <Col
    className="gutter-row"
    lg={
      Object {
        "span": 12,
      }
    }
    md={
      Object {
        "span": 14,
      }
    }
    sm={
      Object {
        "span": 20,
      }
    }
    xl={
      Object {
        "span": 10,
      }
    }
    xs={
      Object {
        "span": 20,
      }
    }
  >
    <Form>
      <Field
        className="top-margin"
        component={[Function]}
        enterButton="Search"
        inputAs={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Search",
            "render": [Function],
          }
        }
        name="search"
        onBlur={[MockFunction]}
        onSearch={[MockFunction]}
        placeholder="Enter movie name"
        size="large"
        suffix={
          <ClearButton
            isVisible="Search..."
            onClick={[MockFunction]}
          />
        }
      />
    </Form>
  </Col>
</Row>
`;
